### ISSUE TITLE
no-any for super-typing only

### ISSUE THREAD

**Issue by wcamarao** [palantir/tslint#3707]
## Feature Suggestion

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.7.1
- __Running TSLint via__: CLI

Currently, __no-any__ rule allows for either:

1. When set to __true__, entirely disabling use of __any__ as in __tslint:all__.
2. When set to __false__, entirely allowing use of __any__ as in __tslint:recommended__.

### Suggestion

Is it reasonable to add an option to the __no-any__ rule so it restricts use of __any__ only when overriding a type definition to a super type, while still allowing for __any__ to be used in other circumstances?

#### tslint.json

```json
{
  "rules": {
    "no-any": [true, "only-super-typing"]
  }
}
```

#### Code

```ts
interface Bar {
    process(data: string)
}

class Foo implements Bar {
    process(data: any) { } // tslint error
}

class Unknown {
    process(data: any) { } // tslint ok
}
```

**Comment by JoshuaKGoldberg** [comment]
@wcamarao this seems like a very esoteric use case, though a nifty idea. Especially given that TypeScript has introduced the often-preferable `unknown`, it doesn't seem likely that this difficult-to-implement _(requires type checker, prone to edge cases)_ option would be worth the maintenance effort.

Can you provide some justification for why it would be a good thing to add?

**Comment by wcamarao** [comment]
@JoshuaKGoldberg per code sample above, this would allow for stronger typing in method signatures between interfaces and classes, i.e. if an interface declares a method must receive a string, classes would only implement that interface if that method receives a string.

Currently, with `no-any` set to `false`, classes are allowed to override method signatures from interfaces, causing a false impression they implement that interface.

On the other hand, setting `no-any` to `true` will disable use of `any` entirely, making it hard to reuse javascript libraries without typedefs.

The proposed new value would allow for `any` to be used for these edge cases, while providing a stronger type check between classes and interfaces.
